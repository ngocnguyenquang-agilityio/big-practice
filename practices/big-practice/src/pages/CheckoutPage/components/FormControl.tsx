// Components
import { Input } from '@components/Input/Input';

// Helpers
import { isEmpty } from '@helpers/utils';

const FormControl = ({
  id = '',
  label = '',
  value,
  variant = 'default',
  placeholder = '',
  onChange,
  errorMessage = '',
}: {
  id: string;
  label: string;
  value: string;
  variant?: 'default' | 'secondary' | 'error' | null | undefined;
  placeholder: string;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  errorMessage?: string;
}) => {
  return (
    <>
      <label
        htmlFor={id}
        className='block mb-2 text-sm font-medium text-neutral-400'
      >
        {label}
      </label>
      <Input
        type='text'
        id={id}
        className='py-3'
        placeholder={placeholder}
        variant={!isEmpty(variant) ? variant : 'default'}
        onChange={onChange}
        value={value}
      />
      {!isEmpty(errorMessage) && <p className='text-sm text-red-500'>{errorMessage}</p>}
    </>
  );
};

export default FormControl;
