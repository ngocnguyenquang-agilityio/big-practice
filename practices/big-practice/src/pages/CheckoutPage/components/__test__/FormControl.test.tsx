// Libs
import { render, RenderResult } from '@testing-library/react';

// Components
import FormControl from '../FormControl';

const props = {
  label: 'Label 1',
  value: 'value1',
  placeholder: 'Placeholder1',
  onChange: jest.fn(),
  errorMessage: 'Error messages',
};

describe('FormControl component', () => {
  let comp: RenderResult;
  beforeEach(() => {
    comp = render(<FormControl {...props} />);
  });

  test('Should render correctly', () => {
    expect(comp).toBeTruthy();
  });

  test('Should match snapshot', () => {
    expect(comp).toMatchSnapshot();
  });

  // TODO: update unit test
});
