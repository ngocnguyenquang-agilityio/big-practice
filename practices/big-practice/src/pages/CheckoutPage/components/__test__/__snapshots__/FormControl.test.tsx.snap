// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormControl component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="block mb-2 text-sm font-medium text-neutral-400"
        for="value1"
      >
        Label 1
      </label>
      <input
        class="w-full rounded-lg border border-neutral-800 bg-transparent px-4 text-sm text-white placeholder:text-neutral-400 py-3"
        id="value1"
        placeholder="Placeholder1"
        type="text"
        value="value1"
      />
      <p
        class="text-sm text-red-500"
      >
        Error messages
      </p>
    </div>
  </body>,
  "container": <div>
    <label
      class="block mb-2 text-sm font-medium text-neutral-400"
      for="value1"
    >
      Label 1
    </label>
    <input
      class="w-full rounded-lg border border-neutral-800 bg-transparent px-4 text-sm text-white placeholder:text-neutral-400 py-3"
      id="value1"
      placeholder="Placeholder1"
      type="text"
      value="value1"
    />
    <p
      class="text-sm text-red-500"
    >
      Error messages
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
